File: ConversationStarterApp/src/ConversationStarterApp.Core/Models/PromptAggregate/Enums/Depth.cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.

namespace ConversationStarterApp.Core;

/// <summary>
/// Represents the depth level of a conversation prompt.
/// </summary>
public enum Depth
{
    /// <summary>
    /// Surface-level, casual questions.
    /// </summary>
    Surface = 0,

    /// <summary>
    /// Moderate depth questions.
    /// </summary>
    Moderate = 1,

    /// <summary>
    /// Deep, meaningful questions.
    /// </summary>
    Deep = 2,

    /// <summary>
    /// Very deep, intimate questions.
    /// </summary>
    Intimate = 3,
}

File: ConversationStarterApp/src/ConversationStarterApp.Web/projects/api/src/lib/models/depth.ts
export enum Depth {
  Surface = 0,
  Moderate = 1,
  Deep = 2,
  Intimate = 3
}

export const DepthLabels: Record<Depth, string> = {
  [Depth.Surface]: 'Surface',
  [Depth.Moderate]: 'Moderate',
  [Depth.Deep]: 'Deep',
  [Depth.Intimate]: 'Intimate'
};
